packages/ui/AddLogoButton/index.tsx

```diff
@@ -0,0 +1,146 @@
+import LoadingSpinner from "../LoadingSpinner";
+import { LogoSettingModal } from "../LogoSettingModal";
+
+interface AddLogoButtonProps {
+  environmentId: string;
+  product: TProduct;
+  type?: string;
+  membershipRole?: TMembershipRole;
+  setLocalProduct?: React.Dispatch<React.SetStateAction<TProduct>>;
+  survey: TSurvey;
+  setIsImageAddedFromAddLogoButton?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+interface LogoChangeEvent extends React.ChangeEvent<HTMLInputElement> {
+  target: HTMLInputElement & EventTarget;
+}
+
+export const AddLogoButton: React.FC<AddLogoButtonProps> = ({
+  environmentId,
+  product,
+  type,
+  membershipRole,
+  setLocalProduct,
+  survey,
+  setIsImageAddedFromAddLogoButton,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [imageUrl, setImageUrl] = useState<string>(product?.brand?.logoUrl);
+  const [isLogoAddEditorOpen, setIsLogoAddEditorOpen] = useState(false);
+
+  const onchangeImageHandler = async (e: LogoChangeEvent) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      await handleUpload(file, environmentId);
+    }
+  };
+
+  const handleUpload = async (file: File, environmentId: string) => {
+    setIsLoading(true);
+    try {
+      const { url, error } = await handleFileUpload(file, environmentId);
+
+      if (error) {
+        toast.error(error);
+        setIsLoading(false);
+        return;
+      }
+
+      setImageUrl(url);
+    } catch (err) {
+      toast.error("Logo upload failed. Please try again.");
+      setIsLoading(false);
+    }
+
+    setIsLoading(false);
+    setIsLogoAddEditorOpen(true);
+  };
+
+  const getStyling = () => {
+    // allow style overwrite is disabled from the product
+    if (!product.styling.allowStyleOverwrite) {
+      return product.styling;
+    }
+
+    // allow style overwrite is enabled from the product
+    if (product.styling.allowStyleOverwrite) {
+      // survey style overwrite is disabled
+      if (!survey.styling?.overwriteThemeStyling) {
+        return product.styling;
+      }
+
+      // survey style overwrite is enabled
+      return survey.styling;
+    }
+
+    return product.styling;
+  };
+
+  return (
+    <>
+      {membershipRole !== "viewer" && !getStyling().hideLogo && (
+        <>
+          {isLoading && (
+            <div className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2">
+              <LoadingSpinner />
+            </div>
+          )}
+          {!product?.brand?.logoUrl ? (
+            <Label
+              htmlFor="addCompanyLogo"
+              className={`${type === "mobile" ? "px-6 py-2" : "px-10 py-4"} rounded-md border border-dashed border-slate-300 bg-slate-100 text-xs text-slate-500  hover:cursor-pointer hover:bg-slate-200/25`}>
+              Add logo here
+              <Input
+                id="addCompanyLogo"
+                className="hidden"
+                type="file"
+                accept="image/jpeg,image/png"
```

Comment:
We'd like to support all image mimes so can you try changing this with `image/*`?

Reply:
well in the ticket it was just png and jpeg,  will change to all